addons/frag/XEH_postInit.sqf

```diff
@@ -1,32 +1,61 @@
 #include "script_component.hpp"
 
-if (isServer) then {
-    GVAR(lastFragTime) = -1;
-    [QGVAR(frag_eh), {_this call FUNC(frago);}] call CBA_fnc_addEventHandler;
-};
-
-["CBA_settingsInitialized", {
-    if (!GVAR(enabled)) exitWith {};
-
-    // Register fire event handler
-    ["ace_firedPlayer", LINKFUNC(fired)] call CBA_fnc_addEventHandler;
-    ["ace_firedNonPlayer", LINKFUNC(fired)] call CBA_fnc_addEventHandler;
-    ["ace_firedPlayerVehicle", LINKFUNC(fired)] call CBA_fnc_addEventHandler;
-    ["ace_firedNonPlayerVehicle", LINKFUNC(fired)] call CBA_fnc_addEventHandler;
-
-    addMissionEventHandler ["EachFrame", {call FUNC(masterPFH)}];
-}] call CBA_fnc_addEventHandler;
-
-// Cache for ammo type configs
-GVAR(cacheRoundsTypesToTrack) = [false] call CBA_fnc_createNamespace;
-
-
-// Debug stuff:
-
-#ifdef DRAW_FRAG_INFO
-[] call FUNC(dev_startTracing);
-#endif
-
-#ifdef DEBUG_MODE_FULL
+[
+    "CBA_settingsInitialized",
+    {
+        if (isServer) then {
+            call FUNC(initBlackList);
+            call FUNC(initMaterialCache);
+            [
+                QEGVAR(common,setShotParents),
+                {
+                    (_this#0) setVariable [QGVAR(shotParent), [_this#1, _this#2]];
+                }
+            ] call CBA_fnc_addEventHandler;
+        };
+
+        #ifdef DEBUG_MODE_DRAW
+        [QGVAR(dev_clearTraces),LINKFUNC(dev_clearTraces)] call CBA_fnc_addEventHandler;
+
+        if (!isServer && hasInterface) then {
+            ["ace_firedPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+            ["ace_firedPlayerNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+            ["ace_firedNonPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+            ["ace_firedPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+            ["ace_firedPlayerVehicleNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+            ["ace_firedNonPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
+        };
+        if (hasInterface) then {
```

Comment:
```suggestion
        if (hasInterface) then {
            if (!isServer) then {
                ["ace_firedPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
                ["ace_firedPlayerNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
                ["ace_firedNonPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
                ["ace_firedPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
                ["ace_firedPlayerVehicleNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
                ["ace_firedNonPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
            };
```
Simplified to avoid a double check.

Reply:
Better yet:
```sqf
#ifdef DEBUG_MODE_DRAW
[QGVAR(dev_clearTraces),LINKFUNC(dev_clearTraces)] call CBA_fnc_addEventHandler;

if (!hasInterface) exitWith {};
if (!isServer) then {
    ["ace_firedPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
    ["ace_firedPlayerNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
    ["ace_firedNonPlayer", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
    ["ace_firedPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
    ["ace_firedPlayerVehicleNonLocal", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
    ["ace_firedNonPlayerVehicle", LINKFUNC(dev_fired)] call CBA_fnc_addEventHandler;
};
GVAR(dev_drawPFEH) = [LINKFUNC(dev_drawTrace), 0] call CBA_fnc_addPerFrameHandler;
[
    "ace_interact_menu_newControllableObject",
    {
        params ["_type"];

        private _action = [
            QGVAR(debugReset),
            "Reset ACE Frag traces",
            "",
            {
                [QGVAR(dev_clearTraces), []] call CBA_fnc_remoteEvent;
                call FUNC(dev_clearTraces);
            },
            {true}
        ] call EFUNC(interact_menu,createAction);
        [
            _type,
            1,
            ["ACE_SelfActions"],
            _action,
            true
        ] call ace_interact_menu_fnc_addActionToClass;
    }
] call CBA_fnc_addEventHandler;
#endif
```